### PR TITLE
Generate documentation via GitHub

### PR DESCRIPTION
**Pull request by LachlanMcKee**
## Description
Updated our github actions to publish the documentation automatically.

I also removed Travis as we no longer use it (also it doesn't seem to be working anymore)

## Check list

- [x] I have updated `whatsnew.md` if required.
- [x] I have updated documentation if required.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,23 +0,0 @@
-language: android
-dist: trusty
-jdk: oraclejdk8
-install: true
-script:
```
Comment (CherryPerry) [inline .travis.yml:5]: Remove `scripts` folder?

File: README.md [hunk 2]
```diff
@@ -1,6 +1,5 @@
 # MVICore
 [![Version](https://jitpack.io/v/badoo/mvicore.svg)](https://jitpack.io/#badoo/mvicore)
-[![Build Status](https://travis-ci.org/badoo/MVICore.svg?branch=master)](https://travis-ci.org/badoo/MVICore)
```
Comment (CherryPerry) [inline README.md:3]: Replace with GitHub actions badge?

Comment (CherryPerry) [reply README.md:3]:
```
[![Build](https://github.com/badoo/MVICore/actions/workflows/build.yml/badge.svg)](https://github.com/badoo/MVICore/actions/workflows/build.yml)
```

Comment (LachlanMcKee) [reply README.md:3]: done!

**Comment by LachlanMcKee** [comment]
@CherryPerry done. good catch!

**Review by CherryPerry — APPROVED** [review]

**Pull request merged**